src/NewProject/ProjectManager/project_manager.cpp

```diff
@@ -49,17 +50,99 @@ void ProjectManager::CreateProject(const ProjectOptions& opt) {
 
   setCurrentFileSet(opt.currentFileSet);
   const QList<filedata> listFile = opt.sourceFileData.fileData;
+
+  // Group and add files to project
+  std::map<std::string, std::vector<filedata>> fileGroups{};
+  std::vector<std::string> keys;
+  // Step through each file to determine our group keys and store the file
+  // under it's group key
   for (const filedata& fdata : listFile) {
-    auto libraries = fdata.m_workLibrary;
-    auto command = libraries.isEmpty() ? QString() : "-work";
+    // Track unique keys in order
+    std::string key = fdata.m_groupName.toStdString();
+    if (std::find(keys.begin(), keys.end(), key) == keys.end()) {
+      keys.push_back(key);
+    }
+    // Store the current file under it's m_groupName key
+    fileGroups[key].push_back(fdata);
+  }
+
+  // Step through the group key and try to combine all the settings
+  for (auto key : keys) {
+    std::string fileListStr{};
+    std::set<std::string> libs{};
+    int language = -1;
+    bool hasLocalFiles = false;
+    bool hasNonLocalFiles = false;
+    bool multipleLanguages = false;
+
+    // Loop through each file in this specific group
+    auto files = fileGroups[key];
+    for (auto fdata : files) {
+      std::string addFilePath{};
+
+      // Track the language
+      if (language == -1) {
+        language = fdata.m_language;
+      } else {
+        if (language != fdata.m_language) {
+          // This might be an error state, can you group files of diff
+          // languages?
+          multipleLanguages = true;
+        }
+      }
 
-    if (LocalToProject == fdata.m_filePath) {
-      setDesignFiles(command, libraries, fdata.m_fileName, fdata.m_language,
-                     false, true);
+      // Split libraries by space and then store any new, unique library names
+      // (the libs std::set will automatically remove dupes)
+      std::vector<std::string> tokens;
+      StringUtils::tokenize(fdata.m_workLibrary.toStdString(), " ", tokens);
```

Comment:
Why you split libs? fdata.m_workLibrary it's a name of one library for one file.

Reply:
Libraries get split in setDesignFiles so i was just preserving that logic by combining all the libs for a group.